File: Java/SETPractice/src/main/java/DesignPattern/FactoryPattern/NotificationFactory.java
package DesignPattern.FactoryPattern;

import java.io.InvalidObjectException;

public class NotificationFactory {

    public INotification GetNotification(NotificationType type) throws InvalidObjectException {
        switch (type) {
            case SMS: return new SMSNotification();
            case PUSH: return new PUSHNotification();
            case EMAIL: return new EmailNotification();
            default: throw new InvalidObjectException("Provide a valid notification type");
        }
    }
}

File: DotNet/SETPractice/DesignPattern/FactoryPattern/NotificationFactory.cs
namespace DesignPattern.FactoryPattern
{
    public class NotificationFactory
    {
        public NotificationFactory() { }

        public INotification GetNotification(NotificationType type)
        {
            switch(type) 
            {
                case NotificationType.SMS: return new SMSNotification();
                case NotificationType.EMAIL: return new EmailNotification();
                case NotificationType.PUSH: return new PushNotification();
                default: throw new InvalidOperationException("Provide a valid Notification type");
            }
        }

    }
}
